### PR TITLE
Rag pattern

### PR DESCRIPTION
**Pull request by esbran**

### CODE REVIEW

File: docs/rag.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# GPT on your Enterprise data
+
+etrieval-Augmented Generation (RAG) is a technique developed by Facebook AI, used to enhance the performance of language models in Natural Language Processing (NLP). It combines the advantages of pre-trained language models like BERT, GPT-2, or T5 with the ability to retrieve and use relevant external knowledge.
```
Comment (victorar) [inline docs/rag.md:3]: Retrieval-Augmented Generation

File: docs/rag.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+The key advantage of RAG is its ability to leverage external knowledge to generate answers, which allows it to provide more detailed and accurate responses compared to models that rely solely on pre-trained knowledge. It's particularly effective in question-answering tasks and can be used in a wide range of applications, including chatbots, customer service, and AI research.
+
+In the context of Azure AI, while Azure OpenAI API does provide an out-of-the-box RAG solution, you can also leverage the suite of AI and machine learning tools to implement such a model. Azure's Machine Learning service provides the infrastructure and tools necessary for training, testing, and deploying machine learning models. You can also use Azure's Document Intelligence API for processing and analyzing text data, and Azure Web App Service for creating conversational AI applications.
+
+## Azure AI Search
+
+Azure AI Search and OpenAI can be paired to create a powerful search and AI solution. OpenAI provides advanced AI models, like GPT-4, which can generate human-like text. Azure AI Search provides sophisticated search capabilities. Here's how they could work together:
```
Comment (victorar) [inline docs/rag.md:15]: which can generate content in a human language.

File: docs/rag.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Querying with OpenAI: When a query is made, it could be processed by OpenAI's GPT-4 to understand the context better. For example, GPT-4 could convert a natural language query into a more structured form that Azure AI Search can understand.
+Search and Retrieval: Azure AI Search retrieves the most relevant results based on the processed query. The results are ranked according to relevance, providing a rich search experience.
+Post-processing with OpenAI: The search results can be further processed by OpenAI's GPT-4 for better presentation. For example, GPT-4 could generate a natural language summary of the search results.
+Monitoring and Maintenance: The performance and usage of the search service can be tracked using Azure Monitor and Azure Log Analytics. The search indices, data sources, and other resources can be managed using the Azure portal, REST APIs, or .NET SDK.
```
Comment (victorar) [inline docs/rag.md:21]: the search indexes

File: docs/rag.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+3. Index Creation: Creating an index in Azure Cognitive Search is fundamental for efficient search operations. The schema defines fields in your data, the data types of these fields, and how these fields should be treated in search operations. The data source points to the location of your data. The indexer crawls over the data source and automatically extracts searchable data and metadata based on the predefined schema.
+4. Query Processing: Azure Cognitive Search supports full-text search queries, filters, facets, and even complex types. It uses a simple query syntax by default, but also supports the Lucene query syntax for more complex search queries.
+5. Azure Function as Application Orchestrator: Azure Functions can be used to orchestrate the flow of data, for example, sending prompts to Azure OpenAI and handling the responses. Azure Functions are serverless and event-driven, meaning they scale automatically to handle incoming traffic.
```
Comment (victorar) [inline docs/rag.md:31]: Would it be better to say: 5. Application Orchestrator (for example, using Azure Functions)

File: docs/rag.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+Querying with OpenAI: When a query is made, it could be processed by OpenAI's GPT-4 to understand the context better. For example, GPT-4 could convert a natural language query into a more structured form that Azure AI Search can understand.
+Search and Retrieval: Azure AI Search retrieves the most relevant results based on the processed query. The results are ranked according to relevance, providing a rich search experience.
+Post-processing with OpenAI: The search results can be further processed by OpenAI's GPT-4 for better presentation. For example, GPT-4 could generate a natural language summary of the search results.
+Monitoring and Maintenance: The performance and usage of the search service can be tracked using Azure Monitor and Azure Log Analytics. The search indices, data sources, and other resources can be managed using the Azure portal, REST APIs, or .NET SDK.
+
+## RAG pattern on Azure
```
Comment (victorar) [inline docs/rag.md:23]: Would it make sense to add numbers (1, 2, ... N) to the picture to match the points below?

**Review by victorar — COMMENTED** [review]
Looks good! pls see my comments.

File: docs/rag.md (hunk 2 above)
Comment (esbran) [reply docs/rag.md:15]: what about human-like content?

File: docs/rag.md (hunk 3 above)
Comment (esbran) [reply docs/rag.md:21]: thats in line 17

File: docs/rag.md (hunk 5 above)
Comment (esbran) [reply docs/rag.md:23]: Working on that. Will be added in the next update.

File: docs/media/highlevel.png
Comment (victorar) [inline docs/media/highlevel.png:1]: This image perhaps needs a bit of updates, as 1) some names are not accurate (like Azure AI Search or Azure OpenAI), and also the picture has numbers (1, 2 and 3) but those numbers do not match to anything in the text, which is a bit misleading.

Comment (esbran) [reply docs/media/highlevel.png:1]: The names are correct at this level, but we can discuss if thats the right level. Azure AI = The suite. OpenAI, if it is Azure or not.

Adding numbers to the text.

**Review by krnese — APPROVED** [review]

**Pull request merged**